crates/evm-storage-proofs/src/client.rs

```diff
@@ -33,20 +33,31 @@ impl EvmClient {
         &self,
         keys: &[&str],
         contract: Address,
-        height: u64,
+        height: Option<u64>,
     ) -> Result<EIP1186AccountProofResponse> {
-        let proof: EIP1186AccountProofResponse = self
-            .provider
-            .get_proof(
-                contract,
-                keys.iter().map(|k| FixedBytes::from_hex(k).unwrap()).collect(),
-            )
-            .block_id(height.into())
-            .await?;
-        Ok(proof)
+        if height.is_some() {
+            let proof: EIP1186AccountProofResponse = self
+                .provider
+                .get_proof(
+                    contract,
+                    keys.iter().map(|k| FixedBytes::from_hex(k).unwrap()).collect(),
+                )
+                .block_id(height.expect("height must be provided").into())
```

Comment:
do we need to `.expect()` on height if we meet the truthy path of `height.is_some()` above?

not sure if we can `.into()` directly here or we have to `unwrap()` but just caught my eye so I thought I'd mention